vms/platformvm/pubsub_filterer.go

```diff
@@ -0,0 +1,89 @@
+package platformvm
+
+import (
+	"github.com/ava-labs/avalanchego/api"
+	"github.com/ava-labs/avalanchego/pubsub"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+)
+
+var _ pubsub.Filterer = &baseFilterer{}
+var _ pubsub.Filterer = &exportFilterer{}
+
+type baseFilterer struct {
+	tx *BaseTx
```

Comment:
Can we update this to instead add the `Addressable` interface to the transaction types? This way instead of creating an individual struct for each different transaction type we can use a single implementation for all of them as long as the type going in supports `Addresses()` and most likely `ID()` as well.

This could then be shared across VMs including coreth.

Reply:
Hi @aaronbuchwald 
Implemented new interface `AddressableID`:

```go
type AddressableID interface {
	Addressable

	// ID returns the unique ID of this data
	ID() ids.ID
}
```

and added it to ExportTx and ImportTx